src/pyrokinetics/gk_code/gkw.py

```diff
@@ -0,0 +1,418 @@
+import copy
+import warnings
+from typing import Any, Dict, Optional
+
+import f90nml
+import numpy as np
+import pint
+from cleverdict import CleverDict
+
+from ..constants import deuterium_mass, electron_mass, pi, sqrt2
+from ..local_geometry import LocalGeometry, LocalGeometryMiller, default_miller_inputs
+from ..local_species import LocalSpecies
+from ..normalisation import SimulationNormalisation as Normalisation
+from ..normalisation import convert_dict, ureg
+from ..numerics import Numerics
+from ..templates import gk_templates
+from ..typing import PathLike
+from .gk_input import GKInput
+
+
+class GKInputGKW(GKInput):
```

Comment:
See lines here to register this correctly.

https://github.com/pyro-kinetics/pyrokinetics/blob/4278eb9f2de1223d9d63dbad706ad09d79029979/src/pyrokinetics/gk_code/gs2.py#L30

This is needed I think

```suggestion
class GKInputGKW(GKInput, FileReader, file_type="GKW", reads=GKInput):
```